src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -27,24 +27,28 @@ class CsvEncoder implements EncoderInterface, DecoderInterface
     const ESCAPE_CHAR_KEY = 'csv_escape_char';
     const KEY_SEPARATOR_KEY = 'csv_key_separator';
     const HEADERS_KEY = 'csv_headers';
+    const FORMULAS_START_CHARACTERS = array('=', '-', '+', '@');
 
     private $delimiter;
     private $enclosure;
     private $escapeChar;
     private $keySeparator;
+    private $escapeFormulas;
 
     /**
      * @param string $delimiter
      * @param string $enclosure
      * @param string $escapeChar
      * @param string $keySeparator
+     * @param bool   $escapeFormulas
      */
-    public function __construct($delimiter = ',', $enclosure = '"', $escapeChar = '\\', $keySeparator = '.')
+    public function __construct($delimiter = ',', $enclosure = '"', $escapeChar = '\\', $keySeparator = '.', $escapeFormulas = true)
```

Comment:
Do we want this to be optional? I mean, shouldn't `true` be the only possible behavior and this be merged as a bugfix in the branch that introduced the encoder? IIRC 3.3 is the lowest actively maintained branch where it exists

Reply:
i'd propose $scapeFormulas to be false by default as this can lead potentially undesired behavior (e.g. adding \t before each negative number - in current implementation!)

Reply:
Setting this argument to `true` by default would be a BC break. The author of the linked article explicitly mentions that you need to decide for each single use case whether or not you can escape formulas.